src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -130,36 +142,85 @@ public bool FormulasAreTheSame(IFormula firstFormula, IFormula secondFormula)
                   return false;
             }
          }
+
          return true;
       }
 
-      public void ExpandDynamicFormulaIn(IModel model)
+      public void ExpandNeighborhoodReferencesIn(IModel model)
       {
-         ExpandDynamicFormulaIn(model.Root);
+         void updatePath(IUsingFormula usingFormula, IFormulaUsablePath path) => updateNeighborhoodReferencingPath(model, path, usingFormula);
+
+         model.Root.GetAllChildren<IUsingFormula>(x => x.Formula.IsReferencingNeighborhood())
+            .Each(x => x.Formula.ObjectPaths.Each(path => updatePath(x, path)));
       }
 
+      private void updateNeighborhoodReferencingPath(IModel model, IFormulaUsablePath formulaUsablePath, IUsingFormula usingFormula)
+      {
+         var pathAsList = formulaUsablePath.ToList();
+         var firstIndex = pathAsList.FindIndex(x => x == NBH);
+         var lastIndex = pathAsList.FindLastIndex(x => x == NBH);
+
+         //Only one occurrence of the marker, this is not a valid path. We do not change anything as it won't be resolved later on
+         if (firstIndex == lastIndex)
+            return;
+
+         //We retrieve the path to first container, and second container 
+         var pathToFirstContainer = pathAsList.Take(firstIndex).ToList();
+         //+1 and -1 in order to skip the NBH tags
+         var pathToSecondContainer = pathAsList.Skip(firstIndex + 1).Take(lastIndex - firstIndex - 1).ToList();
```

Comment:
should we check here, that `pathToSecondContainer `does NOT contain `<NBH>`?
Theoretically we could have (invalid) path like this: `..|<NBH>|..|<NBH>|..|Interstitial|<NBH>|MOLECULE|P1`
Then `pathToSecondContainer `would be `..|<NBH>|..|Interstitial` - which is invalid.

Or will it just automatically crash somewhere later on, so that we don't need this additional check?

Reply:
yes. the path will be invalid and when we validate later, it will say that this is not a correct path. That's why i don't care if there are 3 or 5 tags. I need TWO

Reply:
But in case the path has only `one` NBH-keyword, you exit without throwing the exception (lines 163-165).
I don't quite get the difference between these 2 situations (in both cases, the path is invalid)

Reply:
One nbh , I exit , the path will be checked during resolve step
3nbh. I only remove 2. One will be left. And the resolve step will throw

So those two steps are the same really. We rely on the check coming later to warn about a path that's not valid

Reply:
ok